bot/exts/moderation/voice_gate.py

```diff
@@ -37,208 +34,159 @@
+    async def voice_button(self, interaction: discord.Interaction, button: discord.ui.Button) -> None:
+        """A button that checks to see if the user qualifies for voice verification and verifies them if they do."""
         try:
-            data = await self.bot.api_client.get(f"bot/users/{ctx.author.id}/metricity_data")
-        except ResponseCodeError as e:
-            if e.status == 404:
-                embed = discord.Embed(
-                    title="Not found",
-                    description=(
-                        "We were unable to find user data for you. "
-                        "Please try again shortly, "
-                        "if this problem persists please contact the server staff through Modmail."
-                    ),
-                    color=Colour.red()
+            data = await self.bot.api_client.get(
+                f"bot/users/{interaction.user.id}/metricity_data",
+                raise_for_status=True
+            )
+        except ResponseCodeError as err:
+            if err.response.status == 404:
+                await interaction.response.send_message((
+                    "We were unable to find user data for you. "
+                    "Please try again shortly. "
+                    "If this problem persists, please contact the server staff through ModMail."),
+                    ephemeral=True,
+                    delete_after=GateConf.delete_after_delay,
                 )
-                log.info(f"Unable to find Metricity data about {ctx.author} ({ctx.author.id})")
+                log.info(f"Unable to find Metricity data about {interaction.user} ({interaction.user.id})")
             else:
-                embed = discord.Embed(
-                    title="Unexpected response",
-                    description=(
-                        "We encountered an error while attempting to find data for your user. "
-                        "Please try again and let us know if the problem persists."
-                    ),
-                    color=Colour.red()
+                await interaction.response.send_message((
+                    "We encountered an error while attempting to find data for your user. "
+                    "Please try again and let us know if the problem persists."),
+                    ephemeral=True,
+                    delete_after=GateConf.delete_after_delay,
                 )
-                log.warning(f"Got response code {e.status} while trying to get {ctx.author.id} Metricity data.")
-            try:
-                await ctx.author.send(embed=embed)
-            except discord.Forbidden:
-                log.info("Could not send user DM. Sending in voice-verify channel and scheduling delete.")
-                await ctx.send(embed=embed)
+                log.warning(f"Got response code {err.status} while trying to get {interaction.user.id} Metricity data.")
+            return
 
+        if interaction.user.get_role(Roles.voice_verified):
```

Comment:
Moving this up first will help avoid hitting site if a user has the role already.

Reply:
Scooted up